cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/MapReduceProgramWorkflowRunner.java

```diff
@@ -51,8 +52,8 @@
    */
   @Override
   public Callable<RuntimeContext> create(String name) {
-
-    final MapReduceSpecification mapReduceSpec = workflowSpec.getMapReduce().get(name);
+    ApplicationSpecification spec = workflowProgram.getSpecification();
```

Comment:
should this be called getApplicationSpecification? Otherwise it suggests that it returns a WorkflowSpec or a ProgramSpec.